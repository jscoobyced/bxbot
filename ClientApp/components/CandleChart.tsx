import * as React from 'react';
import { RouteComponentProps } from 'react-router';
import { GoogleCharts } from 'google-charts';
import { DateUtil } from '../utils/date';

interface FetchDataState {
    pairings: Pairing[];
    loading: boolean;
}

export class CandleChart extends React.Component<RouteComponentProps<{}>, FetchDataState> {
    private static ChartData: Array<Pairing>;

    constructor() {
        super();
        this.state = { pairings: [], loading: true };

        fetch('api/Data/pairing/1/5')
            .then(response => response.json() as Promise<Pairing[]>)
            .then(data => {
                CandleChart.ChartData = data;
                this.setState({ loading: false });
            });
    }

    public render() {
        let contents: JSX.Element = <span></span>;
        if (this.state.loading) {
            contents = <p><em>Loading...</em></p>
        }
        else {
            GoogleCharts.load(this.drawChart);
        }

        return <div>
            <h1>BTC on bx.in.th</h1>
            <div id="chart_div"></div>
            {contents}
        </div>;
    }

    private drawChart() {
        const rawData: Array<Pairing> = CandleChart.ChartData;
        const formatedData = new Array<any>();
        const lengthToShow = 96;
        const startIndex = Math.max(0, rawData.length - lengthToShow);
        for (let i = startIndex; i < rawData.length; i++) {
            formatedData.push([
                DateUtil.toHumanHours(new Date(rawData[i].timestamp - 7 * 60 * 60 * 1000)),
                rawData[i].low,
                rawData[i].open,
                rawData[i].close,
                rawData[i].high
            ]);
        }
        var data = GoogleCharts.api.visualization.arrayToDataTable(formatedData, true);

        var options = {
            legend: 'none',
            height: 500,
            candlestick: {
                fallingColor: { strokeWidth: 0, fill: '#a52714' },
                risingColor: { strokeWidth: 0, fill: '#0f9d58' },
                hollowIsRising: true
            },
            colors: ['black']
        };

        var chart = new GoogleCharts.api.visualization.CandlestickChart(document.getElementById('chart_div'));

        chart.draw(data, options);
    }
}

interface Pairing {
    timestamp: number;
    low: number;
    high: number;
    close: number;
    open: number;
}