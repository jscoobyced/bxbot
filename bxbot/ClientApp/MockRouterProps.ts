import { RouteComponentProps } from 'react-router';
import { match } from 'react-router-dom';
import { UnregisterCallback, Href, LocationListener } from 'history';

export class MockRouterProps {
    public static listener: LocationListener = () => { };

    public static getMockRouterProps<P>(data: P) {

        let location = {
            hash: "",
            key: "",
            pathname: "",
            search: "",
            state: {}
        };

        let props: RouteComponentProps<P> = {
            match: {
                isExact: true,
                params: data,
                path: "",
                url: ""
            },
            location: location,
            history: {
                length: 2,
                action: "POP",
                location: location,
                push: () => { },
                replace: () => { },
                go: (num) => { },
                goBack: () => { },
                goForward: () => { },
                block: (t) => {
                    let temp: UnregisterCallback = () => { };
                    return temp;
                },
                createHref: (t) => {
                    var temp: Href = "";
                    return temp;
                },
                listen: (t) => {
                    let temp: UnregisterCallback = () => { };
                    return temp;
                }

            },
            staticContext: {
            }
        };

        return props;
    }
}