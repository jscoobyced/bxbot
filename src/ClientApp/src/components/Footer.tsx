import * as React from 'react';

const version = process.env.VERSION;

export class Footer extends React.Component {
    public render() {
        return <footer>${version}</footer>;
    }
}
