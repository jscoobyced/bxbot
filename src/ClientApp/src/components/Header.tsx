import * as React from 'react';
import { NavLink } from 'react-router-dom';
import { ContentKey } from '../services/ContentKey';
import { ContentService } from '../services/ContentService';

export class Header extends React.Component {

    public render() {
        const websiteName = ContentService.getContent(ContentKey.WebsiteName);
        return <header>
            <img src='/images/wings.svg' className='logo' alt={websiteName + ' logo'} title={websiteName} />
            <span>{websiteName}</span>
            <nav>
                <NavLink to={'/'}>
                    Home
                </NavLink>
                <a href='#' className='menu-space'>&nbsp;</a>
                <a href='#' className='dropdown'>Features</a>
                <div className="dropdown-content">
                <NavLink to={'/candle'}>
                        CandleChart
                    </NavLink>
                    <NavLink to={'/'}>
                        Stuff #1
                    </NavLink>
                    <NavLink to={'/candle'}>
                        Stuff #2
                    </NavLink>
                </div>
                <NavLink to={'/'}>
                    About
                </NavLink>
            </nav>
        </header>;
    }
}