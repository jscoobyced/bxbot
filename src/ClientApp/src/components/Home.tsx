import * as React from 'react';

export class Home extends React.Component {
    public render() {
        return <article>
            <h1>Hello, world!</h1>
        </article>;
    }
}
