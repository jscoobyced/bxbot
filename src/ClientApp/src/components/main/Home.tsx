import * as React from 'react';
import { HomeContent } from '../../services/Models';

export interface HomeProps {
    homeContent: HomeContent;
}

export class Home extends React.Component<HomeProps> {

    public constructor(props: HomeProps) {
        super(props);
    }

    public render() {
        return <article>
            <h1>{this.props.homeContent.articleTitle}</h1>
            <section>{this.props.homeContent.article}</section>
        </article>;
    }
}
