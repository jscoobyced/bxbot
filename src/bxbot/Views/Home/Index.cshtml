@{
    ViewData["Title"] = "Home Page";
}

@section styles {
    <link rel="stylesheet" href="~/dist/style.6e45d99ee3a9c6a2d74d.css?1c5ceb03f16395a5c123">
}
<div id="app-root">Something awesome is coming...</div>

@section scripts {
    <script src="~/dist/vendor.584d115ceaf2f0e28503.js?1c5ceb03f16395a5c123"></script>
    <script src="~/dist/main.27dec449c37ccfcfb1fa.js?1c5ceb03f16395a5c123"></script>
}
