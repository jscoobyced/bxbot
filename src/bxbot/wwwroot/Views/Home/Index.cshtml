@{
    ViewData["Title"] = "Home Page";
}

@section styles {
    <link rel="stylesheet" href="~/dist/style.6e45d99ee3a9c6a2d74d.css?9708653bc96ba89fdd59">
}
<div id="app-root">Something awesome is coming...</div>

@section scripts {
    <script src="~/dist/vendor.584d115ceaf2f0e28503.js?9708653bc96ba89fdd59"></script>
    <script src="~/dist/main.2baf3179b283f2bc2311.js?9708653bc96ba89fdd59"></script>
}
