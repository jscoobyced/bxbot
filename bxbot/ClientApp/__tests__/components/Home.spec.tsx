import * as React from 'react';
import { shallow, mount, render } from 'enzyme';
import { Home } from '../../components/Home';
import { } from 'jest';
import { MockRouterProps } from '../../MockRouterProps';

describe('Home component', () => {
    var routerProps = MockRouterProps.getMockRouterProps<{}>({});
    routerProps.history.block();
    routerProps.history.createHref({});
    routerProps.history.listen(MockRouterProps.listener);
    
    it('should render without throwing an error', () => {
        expect(shallow(<Home match={routerProps.match} history={routerProps.history} location={routerProps.location} />)
            .exists())
            .toBe(true)
    })

    it('contains an <H1> tag', () => {
        expect(shallow(<Home match={routerProps.match} history={routerProps.history} location={routerProps.location} />)
            .find('h1').length)
            .toEqual(1)
    })
});