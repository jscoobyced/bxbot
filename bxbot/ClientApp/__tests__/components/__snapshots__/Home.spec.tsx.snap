// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home component should match snapshot 1`] = `
<div>
  <h1>
    Hello, world!
  </h1>
</div>
`;
